hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3LocatedFileStatus.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a;
+
+import org.apache.hadoop.fs.BlockLocation;
+import org.apache.hadoop.fs.LocatedFileStatus;
+
+/**
+ * {@link LocatedFileStatus} extended to also carry ETag and object version ID.
+ */
+public class S3LocatedFileStatus extends LocatedFileStatus {
```

Comment:
1. should be `S3ALocatedFileStatus`
1. need to think of serialization; sometimes these do get sent around, at least with HDFS

Reply:
> need to think of serialization; sometimes these do get sent around, at least with HDFS

I'll add a serialVersionUID, but is there anything more to do?  My thinking is no based on looking at and comparing to LocatedFileStatus.

Reply:
serialVersionUID added in https://github.com/apache/hadoop/pull/675/commits/807e13b293bd9fee426adc0d67a2572a421cd105

Rename done in https://github.com/apache/hadoop/pull/675/commits/13fab97cf2ee58b3ee25beb2aa5f89540890329d